### PR TITLE
Dynamic request options decorator added

### PR DESCRIPTION
**Pull request by TimurKasatkin**

### CODE REVIEW

**Comment by AnonymousVovus** [comment]
What version of targetJdk will we use, jdk8 or jdk11?
To use jdk8 we should add to build.gradle.kts this block
```kotlin
    tasks {
        withType<JavaCompile> {
            sourceCompatibility = JavaVersion.VERSION_1_8.toString()
            targetCompatibility = JavaVersion.VERSION_1_8.toString()
        }
        withType<KotlinCompile> {
            kotlinOptions {
                jvmTarget = "1.8"
            }
        }
}
```

**Comment by AnonymousVovus** [comment]
gradle-6.5 already released, we can use it.

**Comment by AnonymousVovus** [comment]
Why do we neeed `enum class Projs`?
```kotlin
enum class Projs {

    `jfix-armeria-dynamic-request-options`;

    val dependency get(): String = ":$name"
}
```

**Comment by TimurKasatkin** [comment]
https://github.com/ru-fix/jfix-armeria/blob/master/build.gradle.kts#L186

I have already specified jdk target 8 at gradle build script

**Comment by TimurKasatkin** [comment]
> What version of targetJdk will we use, jdk8 or jdk11?
> To use jdk8 we should add to build.gradle.kts this block
> 
> ```
>     tasks {
>         withType<JavaCompile> {
>             sourceCompatibility = JavaVersion.VERSION_1_8.toString()
>             targetCompatibility = JavaVersion.VERSION_1_8.toString()
>         }
>         withType<KotlinCompile> {
>             kotlinOptions {
>                 jvmTarget = "1.8"
>             }
>         }
> }
> ```

https://github.com/ru-fix/jfix-armeria/blob/master/build.gradle.kts#L186

I have already specified jdk target 8 at gradle build script

**Comment by TimurKasatkin** [comment]
> Why do we neeed `enum class Projs`?
> 
> ```kotlin
> enum class Projs {
> 
>     `jfix-armeria-dynamic-request-options`;
> 
>     val dependency get(): String = ":$name"
> }
> ```

It is for typed references between modules. E.g. like [here](https://github.com/ru-fix/jfix-stdlib/blob/release/3.0/jfix-stdlib-files/build.gradle.kts#L12)

Actually, at the moment there is no need to declare `jfix-armeria-dynamic-request-options`, but I have added it immediately for possible future references.

**Comment by TimurKasatkin** [comment]
> gradle-6.5 already released, we can use it.

good point - now gradle updated to the latest version

File: jfix-armeria-dynamic-request-options/src/main/kotlin/ru/fix/armeria/dynamic/request/options/DynamicRequestOptionsClient.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package ru.fix.armeria.dynamic.request.options
+
+import com.linecorp.armeria.client.Client
+import com.linecorp.armeria.client.ClientRequestContext
+import com.linecorp.armeria.client.HttpClient
+import com.linecorp.armeria.client.SimpleDecoratingClient
+import com.linecorp.armeria.common.HttpRequest
```
Comment (AnonymousVovus) [inline jfix-armeria-dynamic-request-options/src/main/kotlin/ru/fix/armeria/dynamic/request/options/DynamicRequestOptionsClient.kt:7]: `HttpRequest` and `HttpResponse` are unused imports, fix it please

**Review by AnonymousVovus — CHANGES_REQUESTED** [review]

**Pull request merged**